server/app/src/main/java/com/ams/app/repository/StudentRepository.java

```diff
@@ -0,0 +1,16 @@
+package com.ams.app.repository;
+import java.util.List;
+import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.JpaSpecificationExecutor;
+import org.springframework.stereotype.Repository;
+import org.springframework.transaction.annotation.Transactional;
+import com.ams.app.model.Student;
+
+@Repository
+@Transactional
```

Comment:
What's the purpose of using @Transactional?

Reply:
It starts a transaction before the annotated method is executed and commits (or rolls back in case of an exception) the transaction after the method has finished executing.